oci-unit-tests/grafana_test.sh

```diff
@@ -0,0 +1,107 @@
+# shellcheck shell=dash
+
+# shellcheck disable=SC1090
+. "$(dirname "$0")/helper/test_helper.sh"
+
+# cheat sheet:
+#  assertTrue $?
+#  assertEquals 1 2
+#  oneTimeSetUp()
+#  oneTimeTearDown()
+#  setUp() - run before each test
+#  tearDown() - run after each test
+
+# The name of the temporary docker network we will create for the
+# tests.
+readonly DOCKER_PREFIX=oci_grafana_test
+readonly DOCKER_IMAGE="squeakywheel/grafana:edge"
+readonly LOCAL_PORT=63180
+
+oneTimeSetUp() {
+    # Cleanup stale resources
+    tearDown
+}
+
+tearDown() {
+    for container in $(docker ps --filter "name=$DOCKER_PREFIX" --format "{{.Names}}"); do
+        debug "Removing container ${container}"
+        stop_container_sync "${container}"
+    done
+}
+
+docker_run_server() {
+    suffix=$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 8 | head -n 1)
+    docker run \
+       --rm \
+       -d \
+       --name "${DOCKER_PREFIX}_${suffix}" \
+       "$@" \
+       $DOCKER_IMAGE
+}
+
+wait_grafana_container_ready() {
+    local container="${1}"
+    local log="HTTP Server Listen"
+    wait_container_ready "${container}" "${log}"
+}
+
+test_default_config() {
+    debug "Creating all-defaults Grafana container"
+    container=$(docker_run_server -p "$LOCAL_PORT:3000")
+
+    assertNotNull "Failed to start the container" "${container}" || return 1
+    wait_grafana_container_ready "${container}" || return 1
+
+    docker exec "$container" pgrep grafana-server > /dev/null
+    assertTrue $?
```

Comment:
Out of curiosity, should we be using the `assertSomething foo || return 1` idiom whenever we call these assert functions?  That's what I'm doing for all my tests, but I noticed that the others tests don't always do that.

Reply:
I prefer not to have the `|| return 1` and see all the following assertions fail (or succeed). With `|| return 1` the function returns right away, and we get less information. Plus we may have some test-specific teardown code after the asserts, which gets skipped if the function returns early. Still it could make sense in some cases I think.

Reply:
Yeah, that's a good point.  I think we will have to come up with a standard way to write these assertions.